conans/util/files.py

```diff
@@ -154,18 +154,22 @@ def sha256sum(file_path):
 
 
 def _generic_algorithm_sum(file_path, algorithm_name):
+    try:
+        hasher = hashlib.new(algorithm_name)
+    except ValueError:  # FIPS error https://github.com/conan-io/conan/issues/7800
+        hasher = hashlib.new(algorithm_name, usedforsecurity=False)
 
-    with open(file_path, 'rb') as fh:
-        try:
-            m = hashlib.new(algorithm_name)
-        except ValueError:  # FIPS error https://github.com/conan-io/conan/issues/7800
-            m = hashlib.new(algorithm_name, usedforsecurity=False)
-        while True:
-            data = fh.read(8192)
-            if not data:
-                break
-            m.update(data)
-        return m.hexdigest()
+    # broken symlink are allowed to be installed, so they are empty files.
+    if os.path.islink(file_path) and not os.path.exists(os.readlink(file_path)):
+        hasher.update(b'')
+    else:
+        with open(file_path, 'rb') as fh:
```

Comment:
this part is that worries me a lot. what happens for symlinks to files like `/dev/null`, `/dev/urandom`, `/dev/zero`?
I worry it may result in an infinite loop here.

Reply:
Now any symlink is considered an empty file for hashing and we won't open its target. Two problems solved, broken links and device files.